### PR TITLE
update evergreen-ui and add docs for TagInput autocompleteItems prop

### PR DESCRIPTION
**Pull request by zkuzmic**
**Overview**
Update evergreen-ui package in docs and add autocomplete demo for `TagInput`.

**Screenshots (if applicable)**
Screen Shot 2022-02-25 at 11 38 43 AM

**Documentation**
- [ ] Updated Typescript types and/or component PropTypes
- [x] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

**Comment by kamebkj** [comment]
Looking good! One small question and maybe I'm not using it correctly, but when I try to add a new email value in the "Using onAdd and onRemove (advanced)" section, I get an error of invalid email. 
taginput-email

**Comment by zkuzmic** [comment]
> Looking good! One small question and maybe I'm not using it correctly, but when I try to add a new email value in the "Using onAdd and onRemove (advanced)" section, I get an error of invalid email. taginput-email

Oof, looks like that was an existing bug. I can add a fix.

File: docs/documentation/components/tag-input.mdx [hunk 1]
```diff
@@ -122,20 +122,43 @@ function TagPropCallbackTagInputExample() {
   const [values, setValues] = React.useState(['Kauri', 'Willow'])
   return (
     <TagInput
-      tagProps={value => {
+      tagProps={(value) => {
```
Comment (brandongregoryscott) [inline docs/documentation/components/tag-input.mdx:125]: 🤔 Is there a different `prettier` config for the `docs` site? I've seen some churn with the arrow parens + trailing commas in some other files in the repo, but I'm not quite sure where it's coming from 😅 

Not a big deal, but I'd love to get to the bottom of it 😛

**Review by brandongregoryscott — APPROVED** [review]

File: docs/documentation/components/tag-input.mdx (hunk 1 above)
Comment (zkuzmic) [reply docs/documentation/components/tag-input.mdx:125]: Ooo looks like there is a separate config. Didn't realize that.

Comment (brandongregoryscott) [reply docs/documentation/components/tag-input.mdx:125]: I wonder if we break out the root config from `package.json` into a `.prettierrc` file - would it use that for both `src` and `docs`? 🤔 ideally they'd match

**Pull request merged**